controllers/servingruntime_controller.go

```diff
@@ -404,57 +406,55 @@ func runtimeSupportsPredictor(rt *kserveapi.ServingRuntimeSpec, p *api.Predictor
 	return true
 }
 
-// getRuntimesSupportingPredictor returns a list of keys for runtimes that support the predictor p
+// getRuntimesSupportingPredictor returns a map of keys for runtimes that support the predictor p
 //
 // A predictor may be supported by multiple runtimes.
-func (r *ServingRuntimeReconciler) getRuntimesSupportingPredictor(ctx context.Context, p *api.Predictor) ([]types.NamespacedName, error) {
+func (r *ServingRuntimeReconciler) getRuntimesSupportingPredictor(ctx context.Context, p *api.Predictor) (map[string]types.NamespacedName, error) {
```

Comment:
I was thinking here just populate/return the `map[string]struct{}` (just the set of string names), the `NamespacedName`s can be constructed in the calling function.